### DISCUSSION TITLE
Bottom text will dismiss after insert line content

### DISCUSSION THREAD

**Discussion by yangguansen** [sourcelair/xterm.js#4889] [Q&A]
## Details
- Browser and browser version:  Chrome 119
- OS version: Mac
- xterm.js version: 5.2.1

### Steps to reproduce

1. Fill full text to xterm
2.
```
const ESC = '\u001B[';
const OSC = '\u001B]';
const BEL = '\u0007';
const SEP = ';';

xtermRef.current?.write(ESC + '2' + 'A' + ESC + 'G');
xtermRef.current?.write(ESC + '1' + 'L');
xtermRef.current?.write('test insert content');
```
Move cursor up 2 lines, and insert 1 line text, but the previous bottom line will dismiss.
Does xterm have any way to auto flow text content like html?

before insert:
```
line content 1
line content 2
[cursor position]
```

after insert:
```
test insert content [cursor position]
line content 1
```

Expect:
```
test insert content [cursor position]
line content 1
line content 2
```

**Comment by jerch** [comment]
Thats expected behavior and part of the ancient terminal interface stuff I wrote about here https://github.com/xtermjs/xterm.js/discussions/4885#discussioncomment-7598663
